modules/light-clients/06-solomachine/light_client_module.go

```diff
@@ -213,8 +229,8 @@ func (lcm LightClientModule) RecoverClient(ctx sdk.Context, clientID, substitute
 		return errorsmod.Wrap(clienttypes.ErrClientNotFound, substituteClientID)
 	}
 
-	if clientState.GetLatestHeight().GTE(substituteClient.GetLatestHeight()) {
-		return errorsmod.Wrapf(clienttypes.ErrInvalidHeight, "subject client state latest height is greater or equal to substitute client state latest height (%s >= %s)", clientState.GetLatestHeight(), substituteClient.GetLatestHeight())
+	if clientState.Sequence >= substituteClient.Sequence {
+		return errorsmod.Wrapf(clienttypes.ErrInvalidHeight, "subject client state sequence is greater or equal to substitute client state sequence (%d >= %d)", clientState.Sequence, substituteClient.Sequence)
```

Comment:
This can be deleted now?

Reply:
Yeah, its handled at the 02-client level, correct? I can rip it out

Reply:
yup!